imports/plugins/included/product-variant/containers/productGridItemsContainer.js

```diff
@@ -0,0 +1,286 @@
+    }
+  }
+
+  onDoubleClick = () => {
+    const product = this.props.product;
+    const handle = product.__published && product.__published.handle || product.handle;
+
+    Reaction.Router.go("product", {
+      handle: handle
+    });
+
+    Reaction.setActionView({
+      i18nKeyLabel: "productDetailEdit.productSettings",
+      label: "Product Settings",
+      template: "ProductAdmin"
+    });
+
+    if (this.props.isSearch) {
+      this.props.unmountMe();
+    }
+  }
+
+  onClick = (event) => {
+    event.preventDefault();
+    const product = this.props.product;
+
+    if (Reaction.hasPermission("createProduct") && Reaction.isPreview() === false) {
+      if (this.props.isSearch) {
+        let handle = product.handle;
+        if (product.__published) {
+          handle = product.__published.handle;
+        }
+
+        Reaction.Router.go("product", {
+          handle: handle
+        });
+
+        this.props.unmountMe();
+      }
+
+      const isSelected = event.target.closest("li.product-grid-item.active");
+      const list = document.getElementById("product-grid-list");
+
+      if (isSelected) {
+        // If product is already selected, and you are single clicking WITH command key, things whould happen
+        if (event.metaKey || event.ctrlKey || event.shiftKey) {
+          let checkbox = list.querySelector(`input[type=checkbox][value="${product._id}"]`);
+          const items = document.querySelectorAll("li.product-grid-item");
+          const activeItems = document.querySelectorAll("li.product-grid-item.active");
+          const selected = activeItems.length;
+
+          if (event.shiftKey && selected > 0) {
+            const indexes = [
+              Array.prototype.indexOf.call(items, document.querySelector(`li.product-grid-item[id="${product._id}"]`)),
+              Array.prototype.indexOf.call(items, activeItems[0]),
+              Array.prototype.indexOf.call(items, activeItems[selected - 1])
+            ];
+            for (let i = _.min(indexes); i <= _.max(indexes); i++) {
+              checkbox = items[i].querySelector("input[type=checkbox]");
+              if (checkbox.checked === false) {
+                checkbox.checked = true;
+                this.props.itemSelectHandler(checkbox.checked, product._id);
+              }
+            }
+          } else {
+            checkbox.checked = !checkbox.checked;
+            this.props.itemSelectHandler(checkbox.checked, product._id);
+          }
+        }
+      } else {
+        if (event.metaKey || event.ctrlKey || event.shiftKey) {
+          let checkbox = list.querySelector(`input[type=checkbox][value="${product._id}"]`);
+          const items = document.querySelectorAll("li.product-grid-item");
+          const activeItems = document.querySelectorAll("li.product-grid-item.active");
+          const selected = activeItems.length;
+
+          if (event.shiftKey && selected > 0) {
+            const indexes = [
```

Comment:
this looks like a lot of repeated code from the block above. Could we refactor into a common function?